dlt/normalize/items_normalizers.py

```diff
@@ -112,6 +122,45 @@ def __call__(
 
 
 class ParquetItemsNormalizer(ItemsNormalizer):
+    RECORD_BATCH_SIZE = 1000
+
+    def _write_with_dlt_columns(
+        self, extracted_items_file: str,
+        normalize_storage: NormalizeStorage,
+        load_storage: LoadStorage, load_id: str, schema: Schema, root_table_name: str,
+        add_load_id: bool, add_dlt_id: bool
+    ) -> int:
+        import numpy as np
+        new_columns: List[Any] = []
+        if add_load_id:
+            schema.update_table({"name": root_table_name, "columns": {"_dlt_load_id": {"name": "_dlt_load_id", "data_type": "text"}}})
+            new_columns.append((
+                pyarrow.pyarrow.field("_dlt_load_id", pyarrow.pyarrow.string(), nullable=False),
+                lambda batch: np.full(batch.num_rows, load_id)
```

Comment:
can we use categorical type here? https://github.com/huggingface/datasets/issues/5706 (I think Arrow does not have it but take a look). maybe we could at least do that in numpy?

Reply:
There's a dictionary type in arrow which should be the same thing, I'll look into it

Reply:
Seems `pyarrow.dictionary` works: https://github.com/dlt-hub/dlt/pull/704/files#diff-91126b068ef86213d8e95fcc59fa504fc708a9a20bc74c36a165d080a7e26050R134-R138   
The array takes less memory, but we still have to create the list of strings initially. Not sure if any way around that...